sklearn/metrics/regression.py

```diff
@@ -68,12 +80,23 @@ def _check_reg_targets(y_true, y_pred):
         raise ValueError("y_true and y_pred have different number of output "
                          "({0}!={1})".format(y_true.shape[1], y_pred.shape[1]))
 
-    y_type = 'continuous' if y_true.shape[1] == 1 else 'continuous-multioutput'
+    n_outputs = y_true.shape[1]
+    output_weights_options = (None, 'uniform', 'variance')
+    if output_weights not in output_weights_options:
+        output_weights = check_array(output_weights, ensure_2d=False)
+        if n_outputs == 1:
+            raise ValueError("Custom weights are useful only in "
+                             "multi-output cases.")
+        elif n_outputs != len(output_weights):
+            raise ValueError(("There must be equally many custom weights "
+                              "(%d) as outputs (%d).") %
+                             (len(output_weights), n_outputs))
+    y_type = 'continuous' if n_outputs == 1 else 'continuous-multioutput'
 
-    return y_type, y_true, y_pred
+    return y_type, y_true, y_pred, output_weights
 
 
-def _average_and_variance(values, sample_weight=None):
+def _average_and_variance(values, sample_weight=None, axis=None):
```

Comment:
I was thinking more of just writing it out explicitly where needed.

But since it works the way it is right now, in the interest of terminating this PR, I think we should leave it the way it is.